lib/asyn-thread.c

```diff
@@ -554,18 +554,23 @@ static void destroy_async_data(struct Curl_async *async)
 
     if(!done) {
 #ifdef _WIN32
-      if(td->complete_ev)
+      if(td->complete_ev) {
         CloseHandle(td->complete_ev);
-      else
+        td->complete_ev = NULL;
+      }
 #endif
-      Curl_thread_destroy(td->thread_hnd);
+      if(td->thread_hnd != curl_thread_t_null) {
+        Curl_thread_destroy(td->thread_hnd);
+        td->thread_hnd = curl_thread_t_null;
+      }
```

Comment:
That's the correct way: has to check thread_hnd != curl_thread_t_null before calling Curl_thread_destroy.
Curl_thread_destroy does CloseHandle on the thread_hnd, and calling on NULL handle isn't ok. with pthreads it does the check internally.

Reply:
I think that ultimately it would be better to address this in Curl_thread_destroy and get rid of the inconsistency, for example
~~~diff
diff --git a/lib/curl_threads.c b/lib/curl_threads.c
index 222d936..93fa2da 100644
--- a/lib/curl_threads.c
+++ b/lib/curl_threads.c
@@ -131,7 +131,8 @@ curl_thread_t Curl_thread_create(unsigned int (CURL_STDCALL

 void Curl_thread_destroy(curl_thread_t hnd)
 {
-  CloseHandle(hnd);
+  if(hnd != curl_thread_t_null)
+    CloseHandle(hnd);
 }

 int Curl_thread_join(curl_thread_t *hnd)
~~~